### PR TITLE
Model specs

### PR DESCRIPTION
**Pull request by 29td**
### Project Milestone: Blog App: validations and model specs
**_In this PR the following was implemented:_**

**Add the following validations:**

- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

**For the Post model:**

- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

**Add unit specs for all of your models' methods and validations.**

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  #
+  # The different available types are documented in the features, such as in
+  # https://relishapp.com/rspec/rspec-rails/docs
+  config.infer_spec_type_from_file_location!
+
+  # Filter lines from Rails gems in backtraces.
+  config.filter_rails_from_backtrace!
+  # arbitrary gems may also be filtered via:
+  # config.filter_gems_from_backtrace("gem name")
+end
```
Comment (DeliceLydia) [inline spec/rails_helper.rb:63]: - [x] Please check your `RSpec` configuration I am getting an error when I run tests
Screenshot from 2022-11-04 13-15-58

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @29td, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there! highlights for your project:
- Worked on validations
- Followed git workflow
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/rails_helper.rb (hunk 1 above)
Comment (29td) [reply spec/rails_helper.rb:63]: hy @DeliceLydia Thank you for your suggestion 👍

File: spec/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      @user.name = nil
+      expect(@user).to_not be_valid
+    end
+
+    it 'PostCounter must be an integer greater than or equal to zero' do
+      @user.posts_counter = nil
+      expect(@user).to_not be_valid
+    end
+  end
+end
```
Comment (mahbubul14) [inline spec/user_spec.rb:19]: - [ ] Kindly fix the issue for the test. I am getting an error when I run tests. :+1: 

testerror

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @29td 

Great work on making the changes requested by the previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    @user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.',
+                        posts_counter: 0)
+  end
+
+  context 'Validations' do
+    it 'checks for title presence' do
+      post = Post.new(author: @user, text: 'This is my first post', likes_counter: 0, comments_counter: 0)
+      expect(post.valid?).to eq false
+    end
+
```
Comment (thecodechaser) [inline spec/post_spec.rb:14]: - [ ] Please fix this error as requested by the previous reviewer.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @29td👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**